src/shogun/machine/gp/SparseInference.cpp

```diff
@@ -130,8 +129,9 @@ void SparseInference::check_members() const
 {
 	Inference::check_members();
 
-	require(m_inducing_features.num_rows, "Inducing features should not be empty");
-	require(m_inducing_features.num_cols, "Inducing features should not be empty");
+	require(
+	    m_inducing_features != nullptr,
```

Comment:
for shared_ptr you should use [std::shared_ptr::operator bool](https://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool)